docs/release_notes/pending_release.rst

```diff
@@ -82,3 +82,6 @@ Utilities
 
 * Fix ``xaitk_saliency.utils.masking.weight_regions_by_scalar`` to not upcast
   the data type output when ``inv_masks is True``.
+
+* Update ``xaitk_saliency.utils.masking.weight_regions_by_scalar`` to not use
+  matrix multiplication which significantly improve efficiency.
```

Comment:
So, technically the function still uses matrix multiplication (i.e. the `mask * scalar` operation)- maybe instead:

```suggestion
* Update ``xaitk_saliency.utils.masking.weight_regions_by_scalar`` to not use
  fully vectorized operation which significantly improves efficiency.
```

Reply:
And a SO thread that might be somewhat relevant: https://stackoverflow.com/questions/48887461/why-python-loops-over-slices-of-numpy-arrays-are-faster-than-fully-vectorized-op